fbchat/client.py

```diff
@@ -467,8 +467,16 @@ def resetDefaultThread(self):
 
     """
     END DEFAULT THREAD METHODS
-    """
-
+    """ 
+    def removeFriend(self, friend_id=None):
+        """param friend_id: The id of the friend that you want to remove"""
+        payload = {
+            "friend_id": friend_id,
+            "unref": "none",
+            "confirm": "Confirm",
+        }
+        r = self._post(self.req_url.REMOVE_FRIEND,payload)
+        return r.ok
```

Comment:
If `r.ok` is not true, an exception could have been thrown by using `fix_request=False` in `Client._post`. Maybe the return value should be used for detecting whether the removal of the person _actually_ succeeded? I tested four scenarios:
If the client isn't logged in, it's redirected to the login page
If the client is logged in, but hasn't been active in a while, it's presented with an error message (No redirect)
If the operation was successful, the client is redirected to `/removefriend.php?friend_id=[friend_id]&removed&__req=e&_rdr`
And if the friend_id was invalid, it's redirected to `/removefriend.php?friend_id=[friend_id]&removed&err=1431010&__req=e&_rdr`

So, I think we should update `_post` to support the parameter `allow_redirects`, just like `_cleanGet` does, then use that, look at `r['Location']`, and determine whether the redirect url starts with `/removefriend`. If it does, but contains the `err` parameter, we can either throw an error, or just return `False`.

Reply:
I allowed redirects in the _post, but I can't get the location, only with `r.url`. Can I use that?

Reply:
The point of disabling redirects and using `r['Location']` would be to reduce the number of requests that we need to send, but you can start by implementing it without, and just use `r.url`.

Considering testing the error code, though your described method would probably work, I would argue we should use two functions from [`urllib.parse`](https://docs.python.org/3/library/urllib.parse.html), `parse_qs` and `urlparse`, and look at whether the `dict` `parse_qs(urlparse(url).query)` contains an error code.